### DISCUSSION TITLE
How to draw an obstacle/polygon based on tilemap object layer?

### DISCUSSION THREAD

**Discussion by tfkfan** [excaliburjs/Excalibur#2279] [General]
Hi, didn't find some efficient way to implement this, I have a list of polygons parsed from tilemap and It should be drawn on scene

**Comment by eonarheim** [comment]
Hi @tfkfan 

Sorry for the delay, there isn't a built in way yet in the Tiled plugin to draw the polygons automatically. This does sound like a useful feature to add though!

Currently the polygon objects are accessible in the `TiledMap` data structure after loading, the core excalibur Polygon graphic can then be used to draw the polygon.  

<img width="901" alt="image" src="https://user-images.githubusercontent.com/612071/159931815-369671fe-fe43-47ad-ac28-ef00e37e9280.png">


```typescript
const map = new TiledMapResource(...);
const loader = new ex.Loader([map]);
game.start(loader).then(() => {
  ...

  const objectLayer = map.data.getObjectLayerByName('Objects');
  const polygons = objectLayer.getPolygons();
  for (let tiledPolygon of polygons) {
     const offset = ex.vec(tiledPolygon.x, tiledPolygon.y);
     const points = tiledPolygon.polygon;
     const polyGraphic = new ex.Polygon({
        points: points.map(p => ex.vec(p.x, p.y)),
        color: ex.Color.Yellow,
     });

     const star = new ex.Actor({
        pos: offset
     })
     star.graphics.anchor = ex.vec(0, .5);
     star.graphics.use(polyGraphic);
     game.add(star);
  }

...
});

```

Result:

<img width="504" alt="image" src="https://user-images.githubusercontent.com/612071/159932210-0d1ef6e6-5693-4926-8ee3-1e2751024133.png">

**Reply by eonarheim** [reply]
To increase the quality of the polygon graphic you can also increase it's scale artificially, a good start can be your camera's zoom (the reason is it produces a bitmap under the covers before drawing)

```typescript
const polyGraphic = new ex.Polygon({
    points: points.map(p => ex.vec(p.x, p.y).scale(ex.vec(4, 4))),
    color: ex.Color.Yellow
 });

 const star = new ex.Actor({
    pos: offset,
    scale: ex.vec(1/4, 1/4)
 })
```

<img width="561" alt="image" src="https://user-images.githubusercontent.com/612071/159933937-e6f10219-4c4d-4989-a67e-bf91b9f03a10.png">

**Reply by eonarheim** [reply]
I've added a new feature to do this quality scaling. It will be released in v0.26.0 very soon!

https://github.com/excaliburjs/Excalibur/pull/2315
